packages/core/src/codewhispererChat/controllers/chat/controller.ts

```diff
@@ -656,12 +655,24 @@ export class ChatController {
                 request
             )}`
         )
-        let response: GenerateAssistantResponseCommandOutput | undefined = undefined
+        let response: MessengerResponseType | undefined = undefined
         session.createNewTokenSource()
         try {
             this.messenger.sendInitalStream(tabID, triggerID)
             this.telemetryHelper.setConversationStreamStartTime(tabID)
-            response = await session.chat(request)
+            if (isSsoConnection(AuthUtil.instance.conn)) {
+                const { $metadata, generateAssistantResponseResponse } = await session.chatSso(request)
+                response = {
+                    $metadata: $metadata,
+                    message: generateAssistantResponseResponse,
+                }
+            } else {
+                const { $metadata, sendMessageResponse } = await session.chatIam(request as SendMessageRequest)
+                response = {
+                    $metadata: $metadata,
+                    message: sendMessageResponse,
+                }
+            }
             this.telemetryHelper.recordEnterFocusConversation(triggerEvent.tabID)
```

Comment:
what kind of telemetry or error mechanism we can use to understand customer behavior ?